code/modules/mob/living/carbon/death.dm

```diff
@@ -7,8 +7,17 @@
 
 	if(!gibbed)
 		emote("deathgasp")
-
+	
+	handling_hal = 1
+	if(halimage)
+		qdel(halimage)
+	if(halbody)
+		qdel(halbody)
+	if(halitem)
+		qdel(halitem)
+	hal_screwyhud = 0

```

Comment:
Where did you duplicate all this code from/why is there not a proc to clear hallucinations?